crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -69,6 +71,36 @@ impl CommandQueue {
         std::mem::forget(command);
     }
 
+    /// Push a [`BoxedCommand`] onto the queue.
+    #[inline]
+    pub fn push_boxed(&mut self, boxed_command: BoxedCommand) {
+        let size = std::mem::size_of_val(boxed_command.command.deref());
+        let old_len = self.bytes.len();
+
+        self.metas.push(CommandMeta {
+            offset: old_len,
+            func: boxed_command.func,
+        });
+
+        if size > 0 {
+            self.bytes.reserve(size);
+
+            // SAFE: The internal `bytes` vector has enough storage for the
+            // command (see the call the `reserve` above), and the vector has
+            // its length set appropriately.
+            // Also `boxed_command.command` is turned into a raw pointer, so it
+            // does not need to be forgotten.
+            unsafe {
+                std::ptr::copy_nonoverlapping(
+                    Box::into_raw(boxed_command.command) as *const u8,
```

Comment:
Where is the box dropped?

Reply:
This should fix it, freeing the boxed memory.  I'm not using forget here as I'm not sure how it will behave with the Box or with the trait object, if anyone has some pointers on that please share.